diff --git a/arch/arm/boot/dts/am335x-boneblack-wireless.dts b/arch/arm/boot/dts/am335x-boneblack-wireless.dts
index 207d2b63e0eb..9a7983d04b14 100644
--- a/arch/arm/boot/dts/am335x-boneblack-wireless.dts
+++ b/arch/arm/boot/dts/am335x-boneblack-wireless.dts
@@ -109,3 +109,18 @@ ls-buf-en-hog {
 		line-name = "LS_BUF_EN";
 	};
 };
+
+
+&i2c2 {
+    status = "okay";
+
+	tmp102@48 {
+        compatible = "ti,tmp102";
+        reg = <0x48>;
+    };
+};
