diff --git a/arch/arm/boot/dts/am335x-boneblack-wireless.dts b/arch/arm/boot/dts/am335x-boneblack-wireless.dts
index 207d2b63e0eb..00f741e16694 100644
--- a/arch/arm/boot/dts/am335x-boneblack-wireless.dts
+++ b/arch/arm/boot/dts/am335x-boneblack-wireless.dts
@@ -109,3 +109,64 @@ ls-buf-en-hog {
 		line-name = "LS_BUF_EN";
 	};
 };
+
+
+&i2c2 {
+    status = "okay";
+	clock-frequency = <50000>;  // 50kHz
+
+	oled@3c {
+        	compatible = "solomon,ssd1306";
+        	reg = <0x3c>;
+	};
+
+	oled@3c {
+        	compatible = "sinowealth,sh1106";
+        	reg = <0x3c>;
+
+        	width = <128>;
+        	height = <64>;
+        	page-offset = <2>;   // 🔥 IMPORTANT (fixes alignment)
+    	};
+
+	mpu@68 {
+        	compatible = "invensense,mpu6500";   // ✅ FIXED
+        	reg = <0x68>;
+    	};
+
+	ads1115@48 {
+      		compatible = "ti,ads1115";
+	        reg = <0x48>;
+
+        	#address-cells = <1>;
+	        #size-cells = <0>;
+
+	        channel@0 {
+        			reg = <0>;
+        	};
+	        channel@1 {
+            		reg = <1>;
+        	};
+        	channel@2 {
+            		reg = <2>;
+        	};
+	        channel@3 {
+            		reg = <3>;
+        	};
+    	};
+};
