diff --git a/arch/arm/boot/dts/am335x-boneblack-wireless.dts b/arch/arm/boot/dts/am335x-boneblack-wireless.dts
index 207d2b63e0eb..75d03773fe3f 100644
--- a/arch/arm/boot/dts/am335x-boneblack-wireless.dts
+++ b/arch/arm/boot/dts/am335x-boneblack-wireless.dts
@@ -109,3 +109,24 @@ ls-buf-en-hog {
 		line-name = "LS_BUF_EN";
 	};
 };
+
+
+&i2c2 {
+    status = "okay";
+	 clock-frequency = <50000>;   /* 50 kHz */
+	
+	 oled@3c {
+        compatible = "solomon,ssd1306";
+        reg = <0x3c>;
+    };
+};
\ No newline at end of file
